### ISSUE TITLE
Serialization Documentation

### ISSUE THREAD

**Issue by prosa100** [jdermody/brightwire#4]
What is the 'correct' way to serialize/deserialize a neural network?
What is the best way to evaluate a deserialized network?
Thank you.

**Comment by jdermody** [comment]
Nueral network models are annotated with protobuf-net attributes for serialisation. See https://github.com/mgravell/protobuf-net/blob/master/README.md for details. Basically you serialise the model to a stream with the protobuf-net serializer.

Once the graph has been deserialised from a stream it can be used by loading it onto an execution engine, as shown here: https://github.com/jdermody/brightwire/blob/master/SampleCode/IntegerAddition.cs

I'll add an end to end example to the source code in the next week.

**Comment by prosa100** [comment]
Thank you! I found the issue with what I was doing. I had a custom cost function. The deserlizer uses Type.GetType which expects an assembly-qualified name, but the serialize writes only the FullName.
What is the best way to avoid serializing the Backpropagate?

I made a pull request (#6) that fixes this, and also includes a serialization unit test (i look at unit tests for docs sometimes). I love where this project is going!!! 👍
